src/ZODB/FileStorage/fspack.py

```diff
@@ -407,23 +407,43 @@ def pack(self):
 
         self.gc.findReachable()
 
+        def close_files_remove():
+            # blank except: we might be in an IOError situation/handler
+            # try our best, but don't fail
+            try:
+                self._tfile.close()
+            except:
+                pass
+            try:
+                self._file.close()
+            except:
+                pass
+            try:
+                os.remove(self._name + ".pack")
+            except:
+                pass
+            if self.blob_removed is not None:
+                self.blob_removed.close()
+
         # Setup the destination file and copy the metadata.
         # TODO:  rename from _tfile to something clearer.
         self._tfile = open(self._name + ".pack", "w+b")
-        self._file.seek(0)
-        self._tfile.write(self._file.read(self._metadata_size))
+        try:
+            self._file.seek(0)
+            self._tfile.write(self._file.read(self._metadata_size))
 
-        self._copier = PackCopier(self._tfile, self.index, self.tindex)
+            self._copier = PackCopier(self._tfile, self.index, self.tindex)
+
+            ipos, opos = self.copyToPacktime()
+        except IOError:
+            # most probably ran out of disk space or some other IO error
+            close_files_remove()
+            raise  # don't succeed silently
```

Comment:
No: six.reraise() is needed when you want to supply the traceback object explicitly, since

    raise exc_type, exc_value, tb

and

    raise exc_type(exc_value).with_traceback(tb)

require different syntax on 2 and 3.

A simple `raise` works fine on both 2 and 3.